avocado/core/decorators.py

```diff
@@ -73,14 +73,28 @@ def skip(message=None):
     """
     Decorator to skip a test.
     """
-    def decorator(function):
-        if not isinstance(function, type):
-            @wraps(function)
-            def wrapper(*args, **kwargs):  # pylint: disable=W0613
-                raise core_exceptions.TestSkipError(message)
-            function = wrapper
-        function.__skip_test_decorator__ = True
-        return function
+    def decorator(obj):
+        def method_decorator(function):
+            if not isinstance(function, type):
+                @wraps(function)
+                def wrapper(*args, **kwargs):  # pylint: disable=W0613
+                    raise core_exceptions.TestSkipError(message)
+                function = wrapper
+            function.__skip_test_decorator__ = True
+            return function
+
+        def class_decorator(cls):
+            for key in cls.__dict__:
+                value = getattr(cls, key)
+                if callable(getattr(cls, key)):
+                    wrapped = method_decorator(value)
```

Comment:
I don't think I understood the need to set `value` before the `if` statement. Can't you do: 

```
if callable(getattr(cls, key)):
    wrapped = method_decorator(getattr(cls, key))
```
or, using the `value` variable:

```
value = getattr(cls, key)
if callable(value):
    wrapped = method_decorator(value)
```
Maybe I'm missing something here :)

Reply:
Yes, you are right there is no need for setting `value`.